server/index.js

```diff
@@ -1,35 +0,0 @@
-const express = require('express');
-const bodyParser = require('body-parser');
-const cors = require('cors')
-
-const app = express();
-
-app.use( bodyParser.json() );
-app.use(cors())
-
-
-
-let animals = [
-    {
-      name: 'Armadillo',
-      id: '1',
-      imageUrl: 'https://upload.wikimedia.org/wikipedia/commons/thumb/b/b4/Nine-banded_Armadillo.jpg/1200px-Nine-banded_Armadillo.jpg'
-    },
-    {
-      name: 'Tiger',
-      id: '2',
-      imageUrl: 'http://kids.sandiegozoo.org/sites/default/files/2017-06/animal-hero-tiger_0.jpg'
-    }
-  ]
-
-  app.get('/api/zoo', (req, res) => {
-      res.send(animals)
-  }  )
-
-  app.post('/api/newAnimal', (req, res) => {
-      animals.push(req.body.data)
```

Comment:
This would post the animals, but you're not adding an ID. Here's what I would probably do: 
```
const animalToPush = req.body.data;
animalToPush.id = animals.length + 1; //for example the length starts out as 2, so the id will be 3 and so on
animals.push(animalToPush) 

```